### PR TITLE
Netcode shit

### PR DESCRIPTION
**Pull request by LJSonik**
- Added a timeout for game state downloading to prevent definitive join freezes in some cases. The timeout has a minimum value of "jointimeout" and gets higher as the game state grows in size
- If the server tries to kick a joiner who is downloading the game state, they will get a timeout instead, because a regular kick would only happen once the game state has been downloaded
- Added a timeout for player ticcmd packets, again to prevent freezes to happen in some cases
- File/game state downloading is now faster, the speed is controlled by the "downloadspeed" cvar, in packets per tic
- The reason is now properly shown when the server refuses connection
- Changed the default values of "nettimeout" to 10 seconds (previously 15) and "maxsend" to 4 MB (previously 1)
- Added a "noticedownload" cvar that displays a message in the server console when someone is downloading a file

### CODE REVIEW

**Comment by toasterone** [comment]
:+1: I'll note that I don't quite 100% understand everything that's going on, but it's well-commented and I've experienced the code in practice working pleasantly - which is the important part.

**Comment by SteelTitanium** [comment]
👍 No problems here!

File: src/d_clisrv.c [hunk 1]
```diff
@@ -2809,6 +2847,9 @@ void D_ClientServerInit(void)
 	COM_AddCommand("drop", Command_Drop);
 	COM_AddCommand("droprate", Command_Droprate);
 #endif
+#ifdef DEBUGMODE
```
Comment (MonsterIestyn) [inline src/d_clisrv.c]: DEBUGMODE? did you mean _DEBUG? (same for the other two linked places with "DEBUGMODE")

**Comment by MonsterIestyn** [comment]
Left a comment here though master needs to be merged in here first (lol it's happening as I'm speaking anyway)

File: src/d_clisrv.c (hunk 1 above)
Comment (LJSonik) [inline src/d_clisrv.c]: My bad, yes that's what I meant.

**Pull request merged**